javascript/code-challenges/challenges-04.test.js

```diff
@@ -0,0 +1,355 @@
+'use strict';
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 1 - Review
+
+Write a function named updateAnimal that takes in array of animals (strings) and some callback function.
+
+This function should first create a new array. Then iterate over the input array and modify each value based on the callback function provided.
+
+Push each updated animal string into the new array. Return the new array.
+
+HINT: Look at the tests to see how the callback functions are used.
+
+------------------------------------------------------------------------------------------------ */
+
+function upper(str) {
+  return str.toUpperCase();
+}
+
+function lower(str) {
+  return str.toLowerCase();
+}
+
+const updateAnimal = (arr, callback) => {
+  const arr2 = [];
+  arr.forEach(value => arr2.push(callback(value)));
+  return arr2;
+};
+
```

Comment:
Nice work. Using array.map() would tighten this up, though. It automatically creates a new array for you AND pushes all the data you map through into the array.
```js
const updateAnimal = (arr, callback) => {
  arr.map(animal => callback(animal));
}
```